src/borgstore/backends/rclone.py

```diff
@@ -0,0 +1,265 @@
+        self.url = None
+
+    def _rpc(self, command, json_input, **kwargs):
+        """
+        Run the rclone command over the rclone API
+
+        Additional kwargs may be passed to requests
+        """
+        if not self.process:
+            raise BackendMustBeOpen()
+        kwargs = kwargs.copy()
+        if json_input is not None:
+            kwargs["json"] = json_input
+        kwargs["auth"] = (self.user, self.password)
+        r = requests.post(self.url+command, **kwargs)
+        if r.status_code == 404:
+            raise ObjectNotFound(f"Not Found: error {r.status_code}: {r.text}")
+        elif r.status_code != 200:
+            raise BackendError(f"rclone rc command failed: error {r.status_code}: {r.text}")
+        return r.json()
+
+    def create(self):
+        """create (initialize) the rclone storage"""
+        if self.process:
+            raise BackendMustNotBeOpen()
+        with self:
+            try:
+                listing = self.list("")
+            except ObjectNotFound:
+                pass
+            try:
+                next(listing)
+                # Generator has at least one element
+                raise BackendAlreadyExists(f"rclone storage base path already exists: {self.fs}")
+            except StopIteration:
+                # Generator is empty
+                pass
+            self.mkdir("")
+
+    def destroy(self):
+        """completely remove the rclone storage (and its contents)"""
+        if self.process:
+            raise BackendMustNotBeOpen()
+        with self:
+            self._rpc("operations/purge", {
+                "fs": self.fs,
+                "remote": "",
+            })
+
+    def __enter__(self):
+        self.open()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+        return False
+
+    def mkdir(self, name: str) -> None:
+        """create directory/namespace <name>"""
+        self._rpc("operations/mkdir", {
+            "fs": self.fs,
+            "remote": name,
+        })
+
+    def rmdir(self, name: str) -> None:
+        """remove directory/namespace <name>"""
+        self._rpc("operations/rmdir", {
+            "fs": self.fs,
+            "remote": name,
+        })
+
+    def info(self, name) -> ItemInfo:
+        """return information about <name>"""
+        result = self._rpc("operations/stat", {
+            "fs": self.fs,
+            "remote": name,
+            "opt": {
+                "recurse": False,
+                "noModTime": True,
+                "noMimeType": True,
+            },
+        })
+        item = result["item"]
+        if item is None:
+            return ItemInfo(name=name, exists=False, directory=False, size=0)
+        return ItemInfo(
+            name=item["Name"],
+            exists=True,
+            size=item["Size"],
+            directory=item["IsDir"],
+        )
+
+    def load(self, name: str, *, size=None, offset=0) -> bytes:
```

Comment:
guess the code here needs to assert size is None and offset == 0 until partial load is implemented.

Reply:
Oops, forgot about those. Should be easy to implement

Reply:
Also, load needs to `raise BackendMustBeOpen()` if the backend was not opened yet.

Reply:
load needs to raise `ObjectNotFound` for 404.

Reply:
These things will be fixed in the next revision

Reply:
Guess it still needs to raise BackendMustBeOpen if it's not.